ports/esp32/machine_pin.c

```diff
@@ -622,6 +632,10 @@ STATIC const machine_pin_irq_obj_t machine_pin_irq_object[] = {
     {{&machine_pin_irq_type}, GPIO_NUM_44},
     {{&machine_pin_irq_type}, GPIO_NUM_45},
 
+    #elif CONFIG_IDF_TARGET_ESP32S3
+
+    {{&machine_pin_irq_type}, GPIO_NUM_46},
```

Comment:
I don't think this is correct, the case of `CONFIG_IDF_TARGET_ESP32S3` is already covered by the block above.  I think GPIO_NUM_46 should be included for both S2 and S3 (it is in the general pin definition at the top of this file).

So I think this change should be adding GPIO_NUM_46 to the block above, ie:
```c
    ...
    {{&machine_pin_irq_type}, GPIO_NUM_44},
    {{&machine_pin_irq_type}, GPIO_NUM_45},
    {{&machine_pin_irq_type}, GPIO_NUM_46},
    #endif
};
```

Reply:
I'm confused. Yes, IO46 is setup in the pins above, but this change is for IRQ handlers.  There is no IRQs for IO46 on S2 as it's an input only pin, but on the S3, it's a I/O/T - and needs to be added to the IRQ list...  It wasn't there before.

My change on line 637 is adding it to the IRQ list.

Reply:
Even if IO46 is input only, it should still be able to generate (input) IRQs, right?

Even if it's not available on S2 this is still wrong, it needs to be:
```c
    ...
    {{&machine_pin_irq_type}, GPIO_NUM_44},
    {{&machine_pin_irq_type}, GPIO_NUM_45},
    #if CONFIG_IDF_TARGET_ESP32S3
    {{&machine_pin_irq_type}, GPIO_NUM_46},
    #endif
    #endif
};